skyportal/handlers/api/sources.py

```diff
@@ -1846,6 +1850,16 @@ async def get_sources(
                         statement += f"""ORDER BY bool_and(listings.obj_id IS NULL) {sort_order.upper()}"""
                     elif sort_by in NULL_FIELDS:
                         statement += f"""ORDER BY {SORT_BY[sort_by]} {sort_order.upper()} NULLS LAST"""
+                    elif sort_by.startswith("altdata."):
+                        fields = sort_by.split(".")[1:]
+                        altdata_substatement = "altdata->>:altdata_field_0"
```

Comment:
Why did you treat the first element differently from the rest?

Reply:
For nested json data, which is only case where you have more than 1 element, you need to wrap each previous key access of your JSONB to cast it to JSONB again so you can access the next key. Basically once you access a key you, you can't just try access another key in it without saying explicitly that the vale you get when accessing the previous key is a JSONB too. So for the first key I don't have any previous vale to cast to JSONB, hence why I handle it first and then enter the nested case loop for the rest of the elements.

Reply:
Very neat. This is worth a comment in the code for future generations :)

Reply:
Will do! Thanks for the suggestion